cpp/src/binaryop/compiled/util.cpp

```diff
@@ -190,6 +192,26 @@ std::optional<data_type> get_common_type(data_type out, data_type lhs, data_type
 
 bool is_supported_operation(data_type out, data_type lhs, data_type rhs, binary_operator op)
 {
-  return double_type_dispatcher(lhs, rhs, is_supported_operation_functor{}, out, op);
+  return double_type_dispatcher(lhs, rhs, is_supported_operation_functor{}, out, op) ||
+         (is_struct(lhs) && is_struct(rhs) &&
```

Comment:
This logic should be baked into the `is_supported_operation_functor` rather than treated as a special case here.